### PR TITLE
Major JavaDoc cleanups, including grammar, structure, style, and content improvements

### PR DESCRIPTION
**Pull request by dmlloyd**
This works towards #447 and also will make it easier to specify #448, #426, etc.  No behavior changes are specified in this PR as it is meant purely to clean up the formatting and language, to make it easier for upcoming PRs to cleanly specify behavior changes as needed.

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java [hunk 1]
```diff
@@ -42,77 +42,83 @@
...
 public interface ConfigBuilder {
     /**
-     * Add the default config sources appearing on the builder's classpath
-     * including:
-     * <ol>
-     * <li>System properties</li>
-     * <li>Environment properties</li>
-     * <li>/META-INF/microprofile-config.properties</li>
-     * </ol>
+     * Add the <a href="ConfigSource.html#default_config_sources"><em>default configuration sources</em></a>
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: Once this is in the javadoc, I don't think the reference link will work though.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java [hunk 2]
```diff
@@ -42,77 +42,83 @@
...
+     * to the configuration being built.
      *
-     * @return the ConfigBuilder with the default config sources
+     * @return this configuration builder instance
      */
     ConfigBuilder addDefaultSources();
 
     /**
-     * Add the config sources appearing to be loaded via service loader pattern
+     * Add the all configuration sources which can be <a href="ConfigSource.html#discovery">discovered</a> from
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: The reference needs to be the concrete one. For refering its own spec page, it is tricky as the link is not available as yet.

File: api/src/main/java/org/eclipse/microprofile/config/spi/package-info.java [hunk 3]
```diff
@@ -19,31 +19,16 @@
...
+ * The package also contains the class {@link org.eclipse.microprofile.config.spi.ConfigProviderResolver},
+ * which is used to implement the specification itself.
  *
- * <p>The package also contains the class {@link org.eclipse.microprofile.config.spi.ConfigProviderResolver}
- * which is used to pick up the actual implementation.
- * <h2>Usage</h2>
- * <p>This is used to build up a builder and manually add {@link org.eclipse.microprofile.config.spi.ConfigSource ConfigSources}..</p>
- *
- *
- * <ol>
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/package-info.java]: I think the info below is useful. Can you put this in the spec under the 4th bulletin of this [section](https://download.eclipse.org/microprofile/microprofile-config-1.3/microprofile-config-spec.html#configprovider)

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java [hunk 4]
```diff
@@ -42,77 +42,83 @@
...
+     * this configuration builder's {@linkplain #forClassLoader(ClassLoader) class loader}.
      *
-     * @return the ConfigBuilder with the autodiscovered config sources
+     * @return this configuration builder instance
      */
     ConfigBuilder addDiscoveredSources();
 
     /**
-     * Add the converters to be loaded via service loader pattern
+     * Add the all configuration converters which can be <a href="Converter.html#discovery">discovered</a> from
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: add _the_ all => add all

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java [hunk 5]
```diff
@@ -42,77 +42,83 @@
...
     /**
-     * Add the specified {@link Converter} for the given type.
-     * This method does not rely on reflection to determine what type the converter is for
-     * therefore also lambda expressions can be used.
+     * Add the specified {@link Converter} instance for the given type to the configuration being built.
+     * <p>
+     * This method does not rely on reflection to determine the target type of the converter;
+     * therefore, lambda expressions may be used for the converter instance.
+     * <p>
+     * The priority value of custom converters is normally {@code 100}.
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: is normally {@code 100} => defaults to {@code 100} if not specified.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java [hunk 6]
```diff
@@ -26,61 +26,62 @@
...
-     * Create a fresh {@link ConfigBuilder} instance.
-     *
-     * This ConfigBuilder will initially contain no {@link ConfigSource}. The other {@link ConfigSource} will have
-     * to be added manually or discovered by calling {@link ConfigBuilder#addDiscoveredSources()}.
+     * Create a {@link ConfigBuilder} instance for the {@linkplain Thread#getContextClassLoader() current application}.
+     * <p>
+     * The returned configuration builder must initially contain no registered configuration sources.
+     * <p>
+     * The returned configuration builder must initially contain only
+     * <a href="Converter.html#built_in_converters">built in converters</a>.
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java]: see my comments above about the href

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java [hunk 7]
```diff
@@ -32,102 +32,118 @@
...
  *
  * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
  * @author <a href="mailto:gpetracek@apache.org">Gerhard Petracek</a>
  * @author <a href="mailto:emijiang@uk.ibm.com">Emily Jiang</a>
  * @author <a href="mailto:john.d.ament@gmail.com">John D. Ament</a>
- *
  */
 public interface ConfigSource {
+    /**
+     * The name of the default configuration ordinal property, "{@code config_ordinal}".
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/config/spi/ConfigSource.java]: please remove default

**Comment by Emily-Jiang** [comment]
@dmlloyd thanks for the PR! My main concern is the broken link. There are a few other minor comments. Please take a look.

**Comment by Emily-Jiang** [comment]
Can you also change the copyright year to 2020 for the files you udpated?

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java (hunk 1 above)
Comment (dmlloyd) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: It should work: the referenced class is in the same package as this class. This pattern is used extensively within the JDK.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java (hunk 2 above)
Comment (dmlloyd) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: This doesn't refer to a spec page, it creates a link to another JavaDoc element. It even works in IDEs (well IntelliJ at least).

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java [hunk 8]
```diff
@@ -42,77 +42,83 @@
...
+     * this configuration builder's {@linkplain #forClassLoader(ClassLoader) class loader}.
      *
-     * @return the ConfigBuilder with the autodiscovered converters
+     * @return this configuration builder instance
      */
     ConfigBuilder addDiscoveredConverters();
+
     /**
-     * Return the ConfigBuilder for a given classloader
+     * Specify the class loader for which this configuration is being built.
```
[the remark this answers is not in the document]
Comment (dmlloyd) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: Does any implementation return a different builder instance? If not, then no, this method simply tells the builder which class loader to use.

**Comment by dmlloyd** [comment]
Which link is broken? I believe I tested them all.

**Comment by Emily-Jiang** [comment]
> Which link is broken? I believe I tested them all.

Every place with href. e.g. <a href="Discovery.html#discovery">discovered</a> 
How did you test? Javadoc will be published online and it will not be able to find out where the spec pdf lives. See my comments in line above. You can see where the javadoc and pdf published.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java (hunk 1 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: ah. ok. I thought it linked to the spec.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java (hunk 2 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: ack.

**Comment by Emily-Jiang** [comment]
@dmlloyd I am good with your explaination. Can you look at the minor comments I put in line? It will be good if you can get this done today so that this can get into config 1.4.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java (hunk 6 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigProviderResolver.java]: understood now.

File: api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: This sentence reads " specify the class loader for ...". Actually, this method is not about specifying the class loader but creating a config builder for a given class loader.

[the remark this answers is not in the document]
Comment (kenfinnigan) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: I couldn't find anything in the spec that indicates multiple calls to `ConfigBuilder.forClassLoader()` return different instances, so I agree that the new comment makes more sense.

[the remark this answers is not in the document]
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/ConfigBuilder.java]: ok. Let's go with @dmlloyd version!

File: api/src/main/java/org/eclipse/microprofile/config/spi/package-info.java (hunk 3 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/package-info.java]: Let's open a new PR for this.

**Comment by Emily-Jiang** [comment]
@kenfinnigan and I discussed this PR and agreed to merge this. Ken will fix the review comments with a new PR.

**Pull request merged**